src/hatch/publish/index.py

```diff
@@ -3,6 +3,7 @@
 import re
 from typing import Iterable
 
+from hatch.publish.auth import AuthenticationCredentials
```

Comment:
Let's lazily import this, nothing crazy is happening there right now but it might in future